docs/proposals/scheduling/20220701-pod-migration-job.md

```diff
@@ -0,0 +1,393 @@
+  - `PreemptionOptions` decides whether to preempt other Pods if reserved resources failed. The specific details of preemption will be submitted in a separate proposal description in future work, and will not be expanded here for the time being.
+- `DeleteOptions` defines the options of delete operation. Whether to delete a Pod through the `K8s Delete API` or evict a Pod through the `K8s Eviction API` depends on how the user configures the parameters of the `PodMigrationJob Controller`. Users only need to set `DeleteOptions` according to the workload in their own cluster.
+
+#### Migration Job Status
+
+```go
+type PodMigrationJobStatus struct {
+	// PodMigrationJobPhase represents the phase of a PodMigrationJob is a simple, high-level summary of where the PodMigrationJob is in its lifecycle.
+	// e.g. Pending/Running/Failed
+	Phase PodMigrationJobPhase `json:"phase,omitempty"`
+	// Status represents the current status of PodMigrationJob
+	// e.g. ReservationCreated
+	Status string `json:"state,omitempty"`
+	// Reason represents a brief CamelCase message indicating details about why the PodMigrationJob is in this state.
+	Reason string `json:"reason,omitempty"`
+	// Message represents a human readable message indicating details about why the PodMigrationJob is in this state.
+	Message string `json:"message,omitempty"`
+	// Conditions records the stats of PodMigrationJob
+	Conditions []PodMigrationJobCondition `json:"conditions,omitempty"`
+	// NodeName represents the node's name of migrated Pod
+	NodeName string `json:"nodeName,omitempty"`
+	// PodsRef represents the newly created Pods after being migrated
+	PodsRef []v1.ObjectReference `json:"podsRef,omitempty"`
+	// PreemptedPodsRef represents the Pods that be preempted
+	PreemptedPodsRef []v1.ObjectReference `json:"preemptedPodsRef,omitempty"`
+	// PreemptedPodsReservations records information about Reservations created due to preemption
+	PreemptedPodsReservations []PodMigrationJobPreemptedReservation `json:"preemptedPodsReservation,omitempty"`
+}
+
+type PodMigrationJobPreemptedReservation struct {
+	// Namespace represents the namespace of Reservation
+	Namespace string `json:"namespace,omitempty"`
+	// Name represents the name of Reservation
+	Name string `json:"name,omitempty"`
+	// NodeName represents the assigned node for Reservation by scheduler
+	NodeName string `json:"nodeName,omitempty"`
+	// State represents the state of Reservation
+	State string `json:"state,omitempty"`
+	// PreemptedPodRef represents the Pods that be preempted
+	PreemptedPodRef *v1.ObjectReference `json:"preemptedPodRef,omitempty"`
```

Comment:
Does it mean that we can migrate at most one pod in one PodMigrationJob? It differs from the comment.

Reply:
it's typo. fixed.